File: lecture0-30/lecture002/rotatedigits.java
import java.util.Scanner;

public class rotatedigits{
        public static void main(String[] args){
            Scanner scn=new Scanner(System.in);
            int n=scn.nextInt();
            int r=scn.nextInt();
            int digit=0;
            int temp=n;

            while(temp!=0){
                temp/=10;
                digit++;
            }
            r%=digit;
            if(r<0)
            {
                r+=digit;
            }
            int mult=1;
        
            int div=1;
            for(int i=1;i<=digit;i++){
                if(i<=r)
                {
                    mult*=10;
                }
                else{
                    div*=10;
                }
                
            }
            int rem=n%div;
                int oldVal=n/div;

                n=rem*mult+oldVal;
            System.out.print(n);

        }
    }

File: lecture002/rotatedigits.cpp
#include<iostream>
using namespace std;

void rotateDigits(int n, int k)
{
    int temp = n;
    int digits = 0;
    while (temp != 0)
    {
        temp /= 10;
        digits++;
    }

    k %= digits;

    if (k < 0)
    {
        k += digits;
    }

    int div = 1;
    int mult = 1;
    for (int i = 1; i <= digits; i++)
    {
        if (i <= k)
        {
            mult *= 10;
        }
        else
        {
            div *= 10;
        }
    }
    int rem = n % div;
    int oldVal = n / div;

    n = rem * mult + oldVal;
    cout << n;
}

int main()
{
    int n = 2345456;
    int k = 3;
    rotateDigits(n, k);
    return 0;
}